accepted/0000-sbom-command.md

```diff
@@ -0,0 +1,427 @@
+  },
+  "devDependencies": {
+    "@tsconfig/node14": "^14.1.0"
+  }
+}
+```
+
+The complete dependency tree for this project looks like this:
+
+```
+$ npm ls
+
+hello-world@1.0.0 
+├── @tsconfig/node14@14.1.0
+└─┬ debug@4.3.4
+  └── ms@2.1.2
+```
+
+
+### CycloneDX
+
+The proposed CycloneDX SBOM generated for the project above would look like the following:
+
+```json
+{
+  "$schema": "http://cyclonedx.org/schema/bom-1.5.schema.json",
+  "bomFormat": "CycloneDX",
+  "specVersion": "1.5",
+  "serialNumber": "urn:uuid:1b4cd070-3f4c-4f63-965e-4ab302ad7b41",
+  "version": 1,
+  "metadata": {
+    "timestamp": "2023-08-04T21:37:16.639Z",
+    "tools": [
+      {
+        "vendor": "npm",
+        "name": "cli",
+        "version": "9.8.1"
+      }
+    ],
+    "component": {
+      "type": "application",
+      "name": "hello-world",
+      "version": "1.0.0",
+      "bom-ref": "hello-world@1.0.0",
+      "purl": "pkg:npm/hello-world@1.0.0",
+      "properties": [
+        {
+          "name": "cdx:npm:package:path",
+          "value": ""
+        }
+      ]
+    }
+  },
+  "components": [
+    {
+      "type": "library",
+      "name": "node14",
+      "group": "@tsconfig",
+      "version": "1.0.3",
+      "bom-ref": "@tsconfig/node14@1.0.3",
+      "purl": "pkg:npm/%40tsconfig/node14@1.0.3",
+      "properties": [
+        {
+          "name": "cdx:npm:package:path",
+          "value": "node_modules/@tsconfig/node14"
+        },
+        {
+          "name": "cdx:npm:package:development",
+          "value": true
+        }
+      ],
+      "hashes": [
+        {
+          "alg": "SHA-512",
+          "content": "cac4fc9a1762c562..."
+        }
+      ]
+    },
+    {
+      "type": "library",
+      "name": "debug",
+      "version": "4.3.4",
+      "bom-ref": "debug@4.3.4",
+      "purl": "pkg:npm/debug@4.3.4",
+      "properties": [
+        {
+          "name": "cdx:npm:package:path",
+          "value": "node_modules/debug"
+        }
+      ],
+      "hashes": [
+        {
+          "alg": "SHA-512",
+          "content": "3d15851ee494dde0..."
+        }
+      ]
+    },
+    {
+      "type": "library",
+      "name": "ms",
+      "version": "2.1.2",
+      "bom-ref": "ms@2.1.2",
+      "purl": "pkg:npm/ms@2.1.2",
+      "properties": [
+        {
+          "name": "cdx:npm:package:path",
+          "value": "node_modules/ms"
+        }
+      ],
+      "hashes": [
+        {
+          "alg": "SHA-512",
+          "content": "b0690fc7e56332d9..."
+        }
+      ]
+    }
+  ]
+}
+```
+
+### SPDX
+
+The proposed SPDX SBOM generated for the project above would look like the following:
+
+```json
+{
+  "spdxVersion": "SPDX-2.3",
+  "dataLicense": "CC0-1.0",
+  "SPXID": "SPDXRef-DOCUMENT",
+  "name": "hello-world@1.0.0",
+  "documentNamespace": "http://spdx.org/spdxdocs/hello-world-1.0.0-<uuid>",
+  "creationInfo": {
+    "created": "2023-08-04T21:41:02.071Z",
+    "creators": [
+      "Tool: pkg:npm/cli@9.8.1"
+    ]
+  },
```

Comment:
The document needs to define its top-level element, this is the entry point to the SBOM graph:
```suggestion
  },
  "documentDescribes": [
    "SPDXRef-Package-hello-world-1.0.0"
  ],

```